File: task2.js
const express = require('express');
const axios = require('axios');
const app = express();

// Serve a basic HTML form to collect the keyword from the user
app.get('/', (req, res) => {
  res.send(`
    <form method="post" action="/check">
      <input type="text" name="keyword" placeholder="Enter a keyword">
      <button type="submit">Check Keyword</button>
    </form>
  `);
});

// Handle the keyword check
app.post('/check', async (req, res) => {
  const keyword = req.body.keyword;
  const url = 'http://example.com'; // Replace with the desired URL

  try {
    // Fetch the HTML content of example.com
    const response = await axios.get(url);

    // Check if the keyword is present in the HTML content
    if (response.data.includes(keyword)) {
      res.send(`The keyword "${keyword}" was found on ${url}.`);
    } else {
      res.send(`The keyword "${keyword}" was not found on ${url}.`);
    }
  } catch (error) {
    res.send('An error occurred while fetching the page.');
  }
});

app.listen(3000, () => {
  console.log('Server is running on port 3000');
});

File: task2.py
from flask import Flask, request, render_template
import requests

app = Flask(__name__)

@app.route('/', methods=['GET', 'POST'])
def keyword_checker():
    result = None

    if request.method == 'POST':
        keyword = request.form['keyword']
        url = 'http://www.example.com'  # Replace with the URL you want to check.

        try:
            response = requests.get(url)
            if response.status_code == 200:
                content = response.text
                if keyword in content:
                    result = f'The keyword "{keyword}" was found on {url}.'
                else:
                    result = f'The keyword "{keyword}" was not found on {url}.'
            else:
                result = f'Error: Unable to access {url}.'
        except requests.exceptions.RequestException:
            result = 'Error: Unable to access the website.'

    # i manually eddited this to match the repo's structure
    return render_template('task2.html', result=result)

if __name__ == '__main__':
    app.run(debug=True)
